src/actions/socket_actions.js

```diff
@@ -0,0 +1,41 @@
+const sio = require('socket.io-client');
+const ActionTypes = require('../constants').ActionTypes;
+const PayloadSources = require('../constants').PayloadSources;
+const Dispatcher = require('../dispatcher');
+const SocketStore = require('../stores/socket_store');
+
+class SocketActions{
+    init(url){
+        var socket = sio(url);
+        socket.on("connect", this.onConnected);
+
+        Dispatcher.dispatch({
+            actionType: ActionTypes.SOCKET_INITIALIZED,
+            socket: socket
+        });
+        console.log("socket initialized");
```

Comment:
Nit: Remove `console.log`. Also, did the .editiconfig come down with the latest version? It'd be great if we were consistent with spacing (thought not a huge deal).